pkg/tbtc/deposit_sweep.go

```diff
@@ -114,18 +118,7 @@ func newDepositSweepAction(
 }
 
 func (dsa *depositSweepAction) execute() error {
-	walletPublicKeyBytes, err := marshalPublicKey(dsa.wallet().publicKey)
-	if err != nil {
-		return fmt.Errorf("cannot marshal wallet public key: [%v]", err)
-	}
-
-	actionLogger := logger.With(
-		zap.String("wallet", fmt.Sprintf("0x%x", walletPublicKeyBytes)),
-		zap.String("action", dsa.actionType().String()),
-		zap.Uint64("startBlock", dsa.proposalProcessingStartBlock),
-	)
-
-	validateProposalLogger := actionLogger.With(
+	validateProposalLogger := logger.With(
```

Comment:
You refer to the global logger here, not the one held in the action. Also, you won't be able to do `.With` on the `log.StandardLogger` type.

Reply:
Good catch. I fixed it in https://github.com/keep-network/keep-core/pull/3541/commits/cdfcdd71e978489d1d6d02de96c76e017b36e901. I was looking for an interface but it seems like there is none in zap and we have to use `*zap.SugaredLogger` for now. I did it only for the deposit sweep action because the heartbeat action is not creating any new logger.